lua/pac3/core/client/parts/event.lua

```diff
@@ -723,6 +723,99 @@ PART.OldEvents = {
 		end,
 	},
 
+
+	is_no_draw = {
+		callback = function(self, ent)
+			ent = try_viewmodel(ent)
+			return ent:GetNoDraw()
+		end,
+	},
+	
+	is_no_target = {
+		callback = function(self, ent)
+			ent = try_viewmodel(ent)
+			return ent:IsFlagSet( FL_NOTARGET )
+		end,
+	},
+
+
+	// This is a function to grab any and all network variables and try to run them with the correct type, as it's possible to have GetNWInt("Pac", 1) and "GetNWString("Pac" ValueHere") 
+	// This code is designed to try and run things correctly and detect specific types of Network Variables, Since GetNWVar does not have the varable display in the table, but GetNW2var does.
+	// It's also useful if a player doesn't know what the displayed function is as a Networked Variable.
+
+		get_networked = {
+		arguments = {{name = "string"}, {result = "string"}},
+		userdata = {{enums = function()
+			local base_tbl = LocalPlayer():GetNWVarTable()
+			local enum_tbl = {}
+			for k,v in pairs(base_tbl) do
+				enum_tbl[k] = k
+			end
+			return enum_tbl
+		end}},
+		
+		callback = function(self, ent, name, result)
+				ent = try_viewmodel(ent)
+				
+				if ent:GetNWString(tostring(name)) == tostring(result) then
```

Comment:
This isn't a proper type check. We've already talked about this. The value you get depends only on the one that was provided to the set functions during runtime. So, because it's unreliable data, it's not reliable to compare their values to determine the type.

example
] lua_run Entity(1):SetNWBool("varoble", "why")
> Entity(1):SetNWBool("varoble", "why")...
] lua_run_cl print(type(Entity(1):GetNWInt("varoble")))
string

I gave a string to the SetBool and it doesn't matter. it's now a string, not a bool.
I "asked for" an int in the Get and it doesn't matter. it's still a string, never been an int.

So use the isnumber, isbool, isstring functions to decide the type branches

```
local anyvar = ent:GetNWVarTable()[name]
if isstring(anyvar) then
	return self:StringOperator(anyvar, result)
elseif isnumber(anyvar) then
	return self:NumberOperator(anyvar, tonumber(result) or 0)
elseif isbool(anyvar) then
	return anyvar == tobool(result)
end
```

Reply:
I see, I thought my circumstances set it up correctly but I suppose I miss-read what you originally wrote, but I appreciate the correction. You're much more defined at this then I am as I'm not the best when it comes to dealing with GMod's Jank. I do everything in parts over the course of months lol